src/main/java/org/jboss/logmanager/ExtHandler.java

```diff
@@ -429,6 +450,32 @@ public boolean isCallerCalculationRequired() {
         return false;
     }
 
+    @Override
+    protected void reportError(String msg, Exception ex, int code) {
+        super.reportError(msg, ex, code);
+    }
+
+    /**
+     * Report an error using a handler's specific error manager, if any.
+     *
+     * @param handler the handler
+     * @param msg the error message
+     * @param ex the exception
+     * @param code the error code
+     */
+    public static void reportError(Handler handler, String msg, Exception ex, int code) {
+        if (handler != null) {
+            ErrorManager errorManager = handler.getErrorManager();
+            if (errorManager != null) try {
+                errorManager.error(msg, ex, code);
+            } catch (Exception ex2) {
+                // use the same message as the JDK
+                System.err.println("Handler.reportError caught:");
```

Comment:
We should probably use the `StandardOutputStreams` utility here. For something like WildFly where `System.out` and `System.err` are wrapped in a logger, this could cause issues.

Reply:
This replicates the existing JDK behavior, so in WildFly I'd expect it to be caught by the enclosing process logger. In Maven (surefire) I would expect it to be captured late as well so that it doesn't cause the dreaded "streams corrupted" error that `jboss-logmanager` often causes. And in any event, it's an unlikely scenario. But, it's your call.

Reply:
We'll leave it for now and I'll think about how it would affect WildFly.